sqlglot/optimizer/qualify.py

```diff
@@ -62,9 +63,15 @@ def qualify(
         identify: If True, quote all identifiers, else only necessary ones.
 
     Returns:
-        The qualified expression.
+        The qualified expression. If the original expression is a DDL with a `Query` child, then
+        only the child will be qualified and that will be part of the final DDL expression.
     """
     schema = ensure_schema(schema, dialect=dialect)
+
+    original = expression
+    expression = extract_ddl_query(expression)
```

Comment:
why is this being called in qualify and optimize? then it happens twice

Reply:
Because qualify is usually called outside of optimize (e.g. in sqlmesh), and same holds for qualify tables. My reasoning was that this call was low cost and is justifiable to repeat a couple or three times if necessary to ensure these basic rules can work with ddls as well.

Reply:
ok i guess it's low cost

Reply:
@georgesittas if that's the case, we don't need it in optimize, because it happens in qualify right